src/usermod.c

```diff
@@ -866,6 +878,14 @@ static void update_gshadow (void)
 			continue;
 		}
 
+		/*
+		* If rflg+Gflg  is passed in AKA -rG invert is_member, to remove targeted
+		* groups while leaving the user apart of groups not mentioned
+		*/
+		if (Gflg && rflg && ((!is_member && was_member) || (is_member && was_member)) ) {
```

Comment:
Isn't this the same as just

if (Gflg && rflg && was_member) {

Reply:
Please simplify this, and then we can merge.

Reply:
Done